client/state/terms/test/selectors.js

```diff
@@ -65,6 +66,66 @@ describe( 'selectors', () => {
 		} );
 	} );
 
+	describe( 'isRequestingTermsForQueryIgnoringPage()', () => {
+		it( 'should return false if no request exists', () => {
+			const requesting = isRequestingTermsForQueryIgnoringPage( {
+				terms: {
+					queryRequests: {}
+				}
+			}, 2916284, 'categories', {} );
```

Comment:
Yes I have noticed this, corrected it in some places, and left it in others ( like here ) - since as you said potentially there could be a custom `categories` taxonomy.